src/Models/ApplicantModel.php

```diff
@@ -558,4 +558,38 @@ public function appendToAssessmentNotes(int $id, string $assessmentNote): bool
         $query = $this->db->prepare($sql);
         return $query->execute($values);
     }
+
+    public function updateEdaid(array $userInput): bool
+    {
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET edaid=:edaid WHERE id=:id');
+        return $query->execute($userInput);
+    }
+    public function updateGithubUsername(array $userInput): bool
+    {
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET githubUsername=:githubUsername
+         WHERE id=:id');
+        return $query->execute($userInput);
+    }
+
+    public function updateLaptop(array $userInput): bool
+    {
+        #NB:laptop should be 0 or 1 apparently
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET laptop=:laptop WHERE id=:id');
+        return $query->execute($userInput);
+    }
+
+    public function updateUpfront(array $userInput): bool
+    {
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET upfront=:upfront WHERE id=:id');
+        return $query->execute($userInput);
+    }
+
+    public function getFeePaymentMethods(array $userInput): array
+    {
+        $query = $this->db->prepare('SELECT diversitech, edaid, fee, upfront FROM applicants_additional
+                             WHERE id=?');
+        $query->execute([$userInput['id']]);
+        $feePaymentMethod = $query->fetch();
+        return  $feePaymentMethod;
+    }
 }
```

Comment:
for all these methods, it would be better practice to list out separate parameters, rather than having one parameter called userInput. this is because:
- this relies on another part of the system providing the info in the correct format - essentially "tight coupling"
- it makes the code more opaque (harder to tell what the method actually needs to use)
you can see other methods in this model class don't do this, other than the ones that take $applicant which has a million fields